clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java

```diff
@@ -232,28 +273,16 @@ public RecordsIterator(ByteBuffer buffer, CompressionType type, boolean shallow)
         protected LogEntry makeNext() {
             if (innerDone()) {
                 try {
-                    // read the offset
-                    long offset = stream.readLong();
-                    // read record size
-                    int size = stream.readInt();
-                    if (size < 0)
-                        throw new IllegalStateException("Record with size " + size);
-                    // read the record, if compression is used we cannot depend on size
-                    // and hence has to do extra copy
-                    ByteBuffer rec;
-                    if (type == CompressionType.NONE) {
-                        rec = buffer.slice();
-                        int newPos = buffer.position() + size;
-                        if (newPos > buffer.limit())
-                            return allDone();
-                        buffer.position(newPos);
-                        rec.limit(size);
-                    } else {
-                        byte[] recordBuffer = new byte[size];
-                        stream.readFully(recordBuffer, 0, size);
-                        rec = ByteBuffer.wrap(recordBuffer);
+                    LogEntry entry = getNextEntry();
+                    // No more record to return.
+                    if (entry == null)
+                        return allDone();
+
+                    // Convert offset to absolute offset if needed.
+                    if (lastInnerRelativeOffset >= 0) {
+                        long absoluteOffset = entry.offset() - lastInnerRelativeOffset + wrapperRecordOffset;
```

Comment:
Took me a while to wrap my head around this line. It seems like the `lastInnerRelativeOffset` and `wrapperRecordOffset` are constants within the life of this instance, so I'm wondering if we could instead use a single constant (e.g. `absoluteBaseOffset`) for this difference? Then this line would just become: 
```java
long absoluteOffset = absoluteBaseOffset + entry.offset();
```
Which is a lot more readable.